packages/query/src/stacks/stacks-client.ts

```diff
@@ -0,0 +1,305 @@
+import { FtMetadataResponse, NftMetadataResponse } from '@hirosystems/token-metadata-api-client';
+import type {
+  AddressNonces,
+  AddressTransactionsWithTransfersListResponse,
+  BnsGetNameInfoResponse,
+  BnsNamesOwnByAddressResponse,
+  GetRawTransactionResult,
+  MempoolTransaction,
+  MempoolTransactionListResponse,
+  NetworkBlockTimesResponse,
+  ReadOnlyFunctionArgs,
+  ReadOnlyFunctionSuccessResponse,
+  Transaction,
+} from '@stacks/stacks-blockchain-api-types';
+import axios from 'axios';
+
+import { DEFAULT_LIST_LIMIT } from '@leather-wallet/constants';
+import { STX20_API_BASE_URL_MAINNET } from '@leather-wallet/models';
+
+import { Paginated } from '../../types/api-types';
+import { getHiroApiRateLimiter } from '../hiro-rate-limiter';
+import { useLeatherNetwork } from '../leather-query-provider';
+import type {
+  AddressBalanceResponse,
+  ContractInterfaceResponseWithFunctions,
+  NonFungibleTokenHoldingListResult,
+  StacksTxFeeEstimation,
+} from './hiro-api-types';
+import type { Stx20BalanceResponse } from './stx20-api-types';
+
+function HiroAccountsApi(basePath: string) {
```

Comment:
Let's not use this casing. It's not a class anymore, so functions begin with lowercase.

Suggest action based naming: `createHiroAccountApi`

Reply:
Yeah, agree, I'll look at refactoring the BitcoinClient too. I was trying to follow that effort and missed some more changes we can make. Same for comment below.